pedro/src/main/kotlin/dev/nextftc/extensions/pedro/PedroComponent.kt

```diff
@@ -0,0 +1,36 @@
+@file:JvmName("FollowerHolder")
+
+package dev.nextftc.extensions.pedro
+
+import com.pedropathing.follower.Follower
+import com.qualcomm.robotcore.hardware.HardwareMap
+import dev.nextftc.core.components.Component
+import dev.nextftc.core.units.Angle
+import dev.nextftc.core.units.rad
+import dev.nextftc.ftc.ActiveOpMode
+import java.util.function.Supplier
+
+class PedroComponent(private val followerFactory: (hardwareMap: HardwareMap) -> Follower) : Component {
```

Comment:
I think this should be an `object` that is added to an `OpMode` using `PedroComponent.withFollower(...)`;

```suggestion
object PedroComponent : Component {
  lateinit var follower: Follower
    private set

  fun withFollower(follower: Follower) = apply {
    this.follower = follower
  }
```

Reply:
Why do you think it should be a singleton?

Reply:
If `follower` is a globally accessible property (and not a member property), it should be managed by a globally accessible object and not a class.

Reply:
Okay.